core/shared/src/test/scala/fs2/ChunkSpec.scala

```diff
@@ -199,6 +199,11 @@ class ChunkSpec extends Fs2Spec {
     assert(arr2 === Array(0, 0))
   }
 
+  "zipWithIndex andThen toArray" in {
+    val arr: Array[(Int, Int)] = Chunk(0, 0, 0).zipWithIndex.toArray
+    assert(arr === Array((0, 0), (0, 1), (0, 2)))
```

Comment:
We could probably assert `chunk.zipWithIndex.toArray == chunk.toArray.zipWithIndex` and make it a property test :)

Reply:
Good idea :-)